common/djangoapps/third_party_auth/models.py

```diff
@@ -226,7 +236,14 @@ def display_for_login(self):
         Determines whether the provider ought to be shown as an option with
         which to authenticate on the login screen, registration screen, and elsewhere.
         """
-        return bool(self.enabled and self.accepts_logins and self.visible)
+        return bool(self.enabled_for_current_site and self.accepts_logins and self.visible)
+
+    @property
+    def enabled_for_current_site(self):
```

Comment:
Eh, nevermind. I would keep this as `is_enabled_for_current_site()` and have it use `Site.objects.get_current()`. Thanks.